### PR TITLE
util/client.go: add method to specify pod security admission level

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @stlaz @tkashem @soltysh @deads2k

### CODE REVIEW

**Comment by stlaz** [comment]
/retest
infra issues

File: go.mod
Comment (stlaz) [inline go.mod:75]: That's an odd version. Should be at least v0.23.0.

Comment (s-urbaniak) [reply go.mod:75]: indeed, this is what `go mod tidy` generated. let me try to fix this manually.

**Comment by s-urbaniak** [comment]
holding to verify e2e tests

**Comment by s-urbaniak** [comment]
This blocked until we have the newest k8s bump available in origin. The bump was reverted in https://github.com/openshift/origin/pull/26939 because SDN e2e tests broke. There is work in progress in https://github.com/openshift/kubernetes/pull/1228 to unskip those tests. Once merged we can continue here.

File: test/extended/util/client.go
Comment (deads2k) [inline test/extended/util/client.go:111]: non-binding, but if it were me, I would have three methods: `NewCLIWithRestrictedNamespace`, `NewCLIWithBaselineNamespace`, `NewCLIWithPrivilegedNamespace`

**Comment by s-urbaniak** [comment]
blocked by https://github.com/openshift/origin/pull/26955

File: test/extended/util/client.go
Comment (stlaz) [inline test/extended/util/client.go:111]: this should now probably be called just `NewCLIWithPodSecurityLevel` (remove the `Enforce` from the name)

Comment (s-urbaniak) [reply test/extended/util/client.go]: fixed

**Comment by s-urbaniak** [comment]
let's hold this as we pursue a different strategy when this was approved:

- default in this PR to `privileged` for all e2e tests
- explicitly set to `restricted` in a subsequent PR

This reduces the risk of breaking existing e2e tests and allows us to iteratively fix them.

/cc @deads2k @soltysh

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by stlaz** [comment]
/retest-required
irrelevant test failures

File: test/extended/util/client.go
Comment (stlaz) [inline test/extended/util/client.go:11]: order!

Comment (s-urbaniak) [reply test/extended/util/client.go:11]: let's see how CI goes, i don't want to risk another e2e failure rush. i'll address in a followup

**Pull request merged**
